### PR TITLE
Add support for POST/DELETE policy_sets/:policy_set_id/relationships/projects

### PR DESCRIPTION
**Pull request by Netra2104**
## Description

User should be able to add/remove policy sets using POST/DELETE policy_sets/:policy_set_id/relationships/projects in the policy_set resource similar to AddWorkspace and RemoveWorkspace.

## Testing plan
Integration test is added to complete the testing plan

## External links
[Related PR](https://hashicorp.atlassian.net/jira/software/c/projects/TF/boards/593?modal=detail&selectedIssue=TF-6270)

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ envchain go-tfe go test -run TestPolicySetsRemoveWorkspaces -v ./...                                                
=== RUN   TestPolicySetsRemoveWorkspaces
=== RUN   TestPolicySetsRemoveWorkspaces/with_workspaces_provided
=== RUN   TestPolicySetsRemoveWorkspaces/without_workspaces_provided
=== RUN   TestPolicySetsRemoveWorkspaces/with_empty_workspaces_slice
=== RUN   TestPolicySetsRemoveWorkspaces/without_a_valid_ID
--- PASS: TestPolicySetsRemoveWorkspaces (4.10s)
    --- PASS: TestPolicySetsRemoveWorkspaces/with_workspaces_provided (0.30s)
    --- PASS: TestPolicySetsRemoveWorkspaces/without_workspaces_provided (0.00s)
    --- PASS: TestPolicySetsRemoveWorkspaces/with_empty_workspaces_slice (0.00s)
    --- PASS: TestPolicySetsRemoveWorkspaces/without_a_valid_ID (0.00s)
PASS
ok      github.com/hashicorp/go-tfe     5.069s
?       github.com/hashicorp/go-tfe/examples/configuration_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/registry_modules   [no test files]
?       github.com/hashicorp/go-tfe/examples/state_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/users      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]

```

```
envchain go-tfe go test -run TestPolicySetsAddWorkspaces -v ./...
=== RUN   TestPolicySetsAddWorkspaces
=== RUN   TestPolicySetsAddWorkspaces/with_workspaces_provided
=== RUN   TestPolicySetsAddWorkspaces/without_workspaces_provided
=== RUN   TestPolicySetsAddWorkspaces/with_empty_workspaces_slice
=== RUN   TestPolicySetsAddWorkspaces/without_a_valid_ID
--- PASS: TestPolicySetsAddWorkspaces (3.07s)
    --- PASS: TestPolicySetsAddWorkspaces/with_workspaces_provided (0.24s)
    --- PASS: TestPolicySetsAddWorkspaces/without_workspaces_provided (0.00s)
    --- PASS: TestPolicySetsAddWorkspaces/with_empty_workspaces_slice (0.00s)
    --- PASS: TestPolicySetsAddWorkspaces/without_a_valid_ID (0.00s)
PASS
ok      github.com/hashicorp/go-tfe     3.303s
?       github.com/hashicorp/go-tfe/examples/configuration_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/registry_modules   [no test files]
?       github.com/hashicorp/go-tfe/examples/state_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/users      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
     * New `RunPlannedAndSaved` `RunStatus` value
     * New `PlannedAndSavedAt` field in `RunStatusTimestamps`
     * New `RunOperationSavePlan` constant for run list filters
+* Added BETA support for including `projects` relationship and `projects-count` attribute to policy_set on create by @hs26gill [#737](https://github.com/hashicorp/go-tfe/pull/737)
```
Comment (hs26gill) [inline CHANGELOG.md:18]: Seems like this change came from the rebase branch. You can add something like

> Added BETA method AddProjects and RemoveProjects for attaching/detaching policy set to projects

Comment (brandonc) [reply CHANGELOG.md:18]: Also be sure to move the enhancement to the # UNRELEASED section

Comment (Netra2104) [reply CHANGELOG.md]: Fixed the issue!

File: generate_mocks.sh [hunk 2]
```diff
@@ -69,3 +69,4 @@ mockgen -source=agent.go -destination=mocks/agents.go -package=mocks
 mockgen -source=policy_evaluation.go -destination=mocks/policy_evaluation.go -package=mocks
 mockgen -source=project.go -destination=mocks/project_mocks.go -package=mocks
 mockgen -source=registry_no_code_module.go -destination=mocks/registry_no_code_module_mocks.go -package=mocks
+mockgen -source=policy_set.go -destination=mocks/policy_set_mocks.go -package=mocks
```
Comment (brandonc) [inline generate_mocks.sh:72]: This is redundant because it also appears on line 39

**Review by brandonc — APPROVED** [review]
Nonblocking feedback only. Looks great!

**Review by brandonc — APPROVED** [review]
Great!

**Pull request merged**